drift-transport-netty/src/main/java/io/airlift/drift/transport/netty/ReloadableSslContext.java

```diff
@@ -0,0 +1,176 @@
+            Optional<String> privateKeyPassword,
+            long sessionCacheSize,
+            Duration sessionTimeout,
+            List<String> ciphers)
+    {
+        this.trustCertificatesFileWatch = new FileWatch(requireNonNull(trustCertificatesFile, "trustCertificatesFile is null"));
+        requireNonNull(clientCertificatesFile, "clientCertificatesFile is null");
+        this.clientCertificatesFileWatch = clientCertificatesFile.map(FileWatch::new);
+        requireNonNull(privateKeyFile, "privateKeyFile is null");
+        this.privateKeyFileWatch = privateKeyFile.map(FileWatch::new);
+        this.privateKeyPassword = requireNonNull(privateKeyPassword, "privateKeyPassword is null");
+        this.sessionCacheSize = sessionCacheSize;
+        this.sessionTimeout = requireNonNull(sessionTimeout, "sessionTimeout is null");
+        this.ciphers = ImmutableList.copyOf(requireNonNull(ciphers, "ciphers is null"));
+        reload();
+    }
+
+    @Override
+    public SslContext get()
+    {
+        return sslContext.get().getSslContext();
+    }
+
+    public synchronized void reload()
+    {
+        try {
+            // every watch must be called each time to update status
+            boolean trustCertificateModified = trustCertificatesFileWatch.updateState();
+            boolean clientCertificateModified = false;
+            if (clientCertificatesFileWatch.isPresent()) {
+                clientCertificateModified = clientCertificatesFileWatch.get().updateState();
+            }
+            boolean privateKeyModified = false;
+            if (privateKeyFileWatch.isPresent()) {
+                privateKeyModified = privateKeyFileWatch.get().updateState();
+            }
+            if (trustCertificateModified || clientCertificateModified || privateKeyModified) {
+                SslContextBuilder sslContextBuilder = SslContextBuilder.forClient()
+                        .trustManager(trustCertificatesFileWatch.getFile())
+                        .keyManager(
+                                clientCertificatesFileWatch.map(FileWatch::getFile).orElse(null),
+                                privateKeyFileWatch.map(FileWatch::getFile).orElse(null),
+                                privateKeyPassword.orElse(null))
+                        .sessionCacheSize(sessionCacheSize)
+                        .sessionTimeout(sessionTimeout.roundTo(SECONDS));
+                if (!ciphers.isEmpty()) {
+                    sslContextBuilder.ciphers(ciphers);
+                }
+                sslContext.set(new SslContextHolder(sslContextBuilder.build()));
+            }
+        }
+        catch (IOException e) {
+            sslContext.set(new SslContextHolder(new UncheckedIOException(e)));
+        }
+    }
+
+    private static class FileWatch
+    {
+        private final File file;
+        private long lastModified = -1;
+        private long length = -1;
+        private byte[] contents = new byte[0];
```

Comment:
This keeps the contents in memory, which you might want to avoid since they are private keys. This could keep an xxhash instead.